src/test/com/fulcrologic/fulcro/algorithms/normalized_state_helpers_spec.cljc

```diff
@@ -3,33 +3,205 @@
+                                        :person/cars   [[:car/id 1] [:car/id 2]]},
+                                     2 {:person/id     2,
+                                        :person/name   "Dafny",
+                                        :person/age    21,
+                                        :person/spouse [:person/id 2],
+                                        :person/cars   [[:car/id 3]]}},
+                       :root/person [[:person/id 1] [:person/id 2]]})
+
+
+(def denormalized-state-to-one {:root/person {:person/id     1,
+                                              :person/name   "Joe",
+                                              :person/age    24,
+                                              :person/spouse {:person/id   2,
+                                                              :person/name "Dafny",
+                                                              :person/age  21,
+                                                              :person/cars [{:car/id    3,
+                                                                             :car/model "Mercedez"}]}
+                                              :person/cars   [{:car/id 1, :car/model "T-150"}
+                                                              {:car/id 2, :car/model "Ferrari"}]}})
+
+(def denormalized-state-to-many {:root/person [{:person/id     1,
+                                                :person/name   "Joe",
+                                                :person/age    24,
+                                                :person/spouse {:person/id   2,
+                                                                :person/name "Dafny",
+                                                                :person/age  21,
+                                                                :person/cars [{:car/id    3,
+                                                                               :car/model "Mercedez"}]}
+                                                :person/cars   [{:car/id 1, :car/model "T-150"}
+                                                                {:car/id 2, :car/model "Ferrari"}]}
+                                               {:person/id   3,
+                                                :person/name "Billy",
+                                                :person/age  26,
+                                                :person/cars [{:car/id 4, :car/model "Tesla"}]}]})
+
+
+
+
+;;============================================================================
+
+;; TODO Make sure that a path that does not go through idents still resolves correctly (in case of denormalized state)
+
+
+
+(specification "tree-path->db-path"
+  (behavior "In normalized DB, when root key is a to-many list of idents"
+    (let [state normalized-state]
       (assertions
-        (nsh/get-in state [:a :c :value]) => 42)))
-  (behavior "Follows edges from root that are denormalized"
```

Comment:
This is an important behavior and must be supported. You are not required to normalize data, therefore the function should be shown to handle that case without error.

Reply:
Hmm, makes sense as this `get-in` is like an ident-aware variant of the `core/get-in`which should support all the default behavior.

Working on it.